tests/plugins/project/test_extension_helpers.py

```diff
@@ -918,3 +860,131 @@ def _mock_pypi_version(mocker):
         'connect.cli.plugins.project.extension_helpers.requests.get',
         return_value=res,
     )
+
+
+def _setup_extension_validation(mocker):
+    class BasicExtension:
+        def process_asset_purchase_request(self):
+            pass
+
+        def process_tier_config_setup_request(self):
+            pass
+
+        def execute_product_action(self):
+            pass
+
+        def process_product_custom_event(self):
+            pass
+
+    mocker.patch(
+        'connect.cli.plugins.project.extension_helpers.is_bundle',
+        return_value=False,
+    )
+    mocker.patch.object(
+        EntryPoint,
+        'load',
+        return_value=BasicExtension,
+    )
+    mocker.patch(
+        'connect.cli.plugins.project.extension_helpers.pkg_resources.iter_entry_points',
+        side_effect=(
+            iter([EntryPoint('extension', 'connect.eaas.ext')]),
+            iter([EntryPoint('extension', 'connect.eaas.ext')]),
+        ),
+    )
+
+
+def test_validate_variables_key(
+    mocker,
+    capsys,
+    mocked_extension_descriptor,
+):
+    _setup_extension_validation(mocker)
+    _mock_pypi_version(mocker)
+    mocked_extension_descriptor['variables'] = [
+        {'name': 'foo_var', 'value': 'foo_value'},
```

Comment:
IMHO maybe it is better to use realistic keys even though they are not validated (initial_value instead of value).